Firmware/first_lay_cal.cpp

```diff
@@ -10,6 +10,32 @@
 #include "cmdqueue.h"
 #include "mmu2.h"
 #include <avr/pgmspace.h>
+#include <math.h>
+
+//! @brief Count extrude length
+//!
+//! @param layer_height layer height in mm
+//! @param extrusion_width extrusion width in mm
+//! @param extrusion_length extrusion length in mm
+//! @return filament length in mm which needs to be extruded to form line
+__attribute__((noinline))
+static constexpr float count_e(float layer_height, float extrusion_width, float extrusion_length, float filament_diameter=1.75f)
```

Comment:
Minor nitpick but I would write `static constexpr float __attribute__((noinline))` to be consistent with other code in the project. The attribute is applied currently, so this isn't an issue :)

Reply:
@gudnimg No worries, I will change it,  I am fond of consistency myself.